sdk/compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Env.hs

```diff
@@ -193,20 +185,11 @@ instance SomeErrorOrWarning UnwarnableError where
 
 instance SomeErrorOrWarning WarnableError where
   diagnosticWithContextF getter err = do
-    shouldSwap <- getDiagnosticSwapIndicatorF getter
-    if shouldSwap (Left err)
-       then do
-        ctx <- view $ getter . locCtx
-        modify' (WContext ctx (WErrorToWarning err) :)
-       else do
-        throwWithContextFRaw getter (EWarnableError err)
-
-instance SomeErrorOrWarning Warning where
-  diagnosticWithContextF getter warning = do
-    shouldSwap <- getDiagnosticSwapIndicatorF getter
-    if shouldSwap (Right warning)
-       then do
-        throwWithContextFRaw getter (EWarningToError warning)
-       else do
-        ctx <- view $ getter . locCtx
-        modify' (WContext ctx warning :)
+    status <- getWarningStatusF getter err
+    case status of
+      AsError -> throwWithContextFRaw getter (EWarnableError err)
+      AsWarning -> warnWithContextFRaw getter (WErrorToWarning err)
+      Hidden -> pure ()
+
+instance SomeErrorOrWarning StandaloneWarning where
+  diagnosticWithContextF = warnWithContextF
```

Comment:
Why does this no longer have logic to upgrade to an error here?

Reply:
It's to clean up and dedupe some logic.

It now defers to warnWithContextF, which always takes an unambiguous `StandaloneWarning` and uplifts it to `Warning` before emitting it with `warnWithContextFRaw`. The raw functions `warnWithContextFRaw` and `throwWithContextFRaw` will *always* warn or throw, with no chance to switch. The diagnosticWithContextF function, on the other hand, will choose. When receiving a StandaloneWarning it will always turn that to a warning, when receiving a standalone error it will always turn that to an error, only when receiving a WarnableError will it decide which to dispatch to.